core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -255,9 +255,14 @@ private Set<QualifiedTablePrefix> calculatePrefixesWithSchemaName(
         }
 
         Session session = ((FullConnectorSession) connectorSession).getSession();
-        return listSchemaNames(session)
+        List<QualifiedTablePrefix> schemaNames = listSchemaNames(session).collect(toImmutableList());
+        Set<QualifiedTablePrefix> prefixes = schemaNames.stream()
                 .filter(prefix -> predicate.get().test(schemaAsFixedValues(prefix.getSchemaName().get())))
                 .collect(toImmutableSet());
+        if (prefixes.size() == schemaNames.size()) {
+            return ImmutableSet.of(new QualifiedTablePrefix(catalogName));
```

Comment:
We should use `io.trino.connector.informationschema.InformationSchemaMetadata#defaultPrefixes`